src/components/menu/menu-profile.js

```diff
@@ -0,0 +1,77 @@
+// @flow
+import * as React from 'react';
+// Components
+import {
+  ListProfile as StyledListProfile,
+  ListItemProfile as StyledListItemProfile,
+  ProfileImgContainer as StyledProfileImgContainer,
+  ProfileImg as StyledProfileImg,
+  ProfileLabelsContainer as StyledProfileLabelsContainer,
+  ProfileTitle as StyledProfileTitle,
+  ProfileSubtitle as StyledProfileSubtitle,
+  ProfileBody as StyledProfileBody,
+} from './styled-components';
+import {mapStyletronProps} from './utils';
+import {getOverride} from '../../helpers/overrides';
+// Types
+import type {StatelessMenuProfilePropsT} from './types';
+
+export default function MenuProfile({
+  items,
+  getProfileItemLabels,
+  getProfileItemImg,
+  getProfileItemImgText,
+  getRequiredItemProps = (item, index) => ({key: String(index)}),
+  rootRef = React.createRef(),
+  overrides = {},
+}: StatelessMenuProfilePropsT) {
+  const ListProfile = getOverride(overrides.ListProfile) || StyledListProfile;
+  const ListItemProfile =
+    getOverride(overrides.ListItemProfile) || StyledListItemProfile;
+
+  const ProfileImgContainer =
+    getOverride(overrides.ProfileImgContainer) || StyledProfileImgContainer;
+  const ProfileImg = getOverride(overrides.ProfileImg) || StyledProfileImg;
+
+  const ProfileLabelsContainer =
+    getOverride(overrides.ProfileLabelsContainer) ||
+    StyledProfileLabelsContainer;
+  const ProfileTitle =
+    getOverride(overrides.ProfileTitle) || StyledProfileTitle;
+  const ProfileSubtitle =
+    getOverride(overrides.ProfileSubtitle) || StyledProfileSubtitle;
+  const ProfileBody = getOverride(overrides.ProfileBody) || StyledProfileBody;
+
+  return (
+    <ListProfile $ref={rootRef}>
+      {items.map((item, index) => {
+        const requiredProps = getRequiredItemProps(item, index);
+        // $FlowFixMe
+        const {key, ...itemProps} = mapStyletronProps(requiredProps);
+        const ItemImg = getProfileItemImg(item);
+        const {title, subtitle, body} = getProfileItemLabels(item);
+        return (
+          // Need to be explicit with `key` otherwise eslint throws error?
+          <ListItemProfile key={key} {...itemProps}>
+            <ProfileImgContainer>
+              {ItemImg &&
+                (typeof ItemImg === 'string' ? (
```

Comment:
How do we better detect that the user is passing in a React component vs. just an Image URL here? There are libraries out there that we could use, or we could roll our own?

Reply:
React.isValidElement(component)

Reply:
I think that only tests for elements (i.e. after JSX has been parsed), in this case we're expecting a component class/function.

Reply:
This seems like a good enough check to me? `typeof ItemImg === 'function'` might be an okay check for component since that covers stateless functional components and classes, but I've had trouble getting flow to accept that on occasion.

Reply:
look here, it's oficial
https://www.npmjs.com/package/react-is

Reply:
Nice, hadn't seen that

Reply:
Ooh, hadn't seen that either. Thanks, @blackswanny

Reply:
Hm, with `react-is`, this code seems overly verbose. Any potential issue with leaving this as-is and let React errors take care of the last mile (i.e. if users don't pass in a valid React component React will just error out anyway).